src/lib/storage/storage_manager.rb

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yastx"
+require "storage"
+
+module Yast
+  module Storage
+    #
+    # Singleton class for the libstorage object.
+    #
+    # You can simply use StorageManager.instance to use it and create a
+    # libstorage instance if there isn't one yet; this will use common default
+    # parameters for creating it.
+    #
+    # If you need special parameters for creating it, you can use
+    # StorageManager.create_instance with a custom ::Storage::Environment
+    # (notice the difference between the global ::Storage namespace which is
+    # libstorage and this Yast::Storage namespace).
+    #
```

Comment:
Note to self (will comment further after reading the whole code).

I'm still in the early stage of reading the code, but this usage of the singleton patterns looks slightly weird to me.
In general, I'm not a fan of the pattern, but making it possible to initialize the singleton object in different ways (with and without params), makes it worse.